server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java

```diff
@@ -251,35 +238,70 @@ protected void internalStart() {
     } catch (Exception e) {
       throw new RuntimeException(e);
     }
-    port = connector.getLocalPort();
 
-    LOG.info("Service listening on port {}.", getPort());
+    if (null != serverConnector && null != handlerList) {
+      port = serverConnector.getLocalPort();
+      LOG.info("Service listening on port {}.", getPort());
 
-    // Set the information about the address for this server
-    try {
-      this.handler.setServerRpcMetadata(createRpcServerMetadata(connector));
-    } catch (UnknownHostException e) {
-      // Failed to do the DNS lookup, bail out.
-      throw new RuntimeException(e);
+      // Set the information about the address for this server
+      try {
+        this.handler.setServerRpcMetadata(createRpcServerMetadata(serverConnector));
+      } catch (UnknownHostException e) {
+        // Failed to do the DNS lookup, bail out.
+        throw new RuntimeException(e);
+      }
+    } else if (0 == server.getConnectors().length) {
+      LOG.warn("No server connectors have been configured for this Avatica server.");
```

Comment:
This sounds like it should be thrown as a RuntimeException to me. If there are no connectors, this HttpServer couldn't possibly be doing what it's intended to do.